### DISCUSSION TITLE
shared state build error

### DISCUSSION THREAD

**Discussion by testforvln** [DioxusLabs/dioxus#1373] [General]
**Problem**

When using shared state, the hello-dioxus project failed compiling.

**Steps To Reproduce**

[hello-dioxus.zip](https://github.com/DioxusLabs/dioxus/files/12376167/hello-dioxus.zip)
-  unzip the project files "hello-dioxus.zip"
-  cargo build

**Expected behavior**

error[E0521]: borrowed data escapes outside of function 

**Screenshots**
Error like below:
 |
52 | #[inline_props]
   | --------------- lifetime `'a` defined here
53 | fn NavBar(cx: Scope) -> Element {
   |           -- `cx` is a reference that is only valid in the function body
54 |     let is_logged_in = use_shared_state::<IsLoggedIn>(cx).unwrap();
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                        |
   |                        `cx` escapes the function body here
   |                        argument requires that `'a` must outlive `'static`

For more information about this error, try `rustc --explain E0521`.

**Environment:**

 - Dioxus version: [v0.4]
 - Rust version: [1.71.1]
 - OS info: [Win10]
 - App platform: [`web`]

**Comment by ealmloff** [comment]
You need to clone the use_shared state before moving it into a future.

```rust
let state = use_shared_state(cx, || false);
cx.spawn({
    to_owned![state];
    async move {
        // use state in the future now that it has been cloned and moved
    }
})
```

(Converting this to a discussion because it seems more like a help question than issue/feature request with Dioxus)

**Reply by testforvln** [reply]
Thanks! It solves my problem.
